Hast.Communication/Services/IMemoryResourceChecker.cs

```diff
@@ -0,0 +1,21 @@
+﻿using Hast.Common.Interfaces;
+using Hast.Layer;
+using Hast.Transformer.Abstractions.SimpleMemory;
+using System;
+
+namespace Hast.Communication.Services
+{
+    /// <summary>
+    /// Checks if the memory resources can fit the <see cref="SimpleMemory"/> before sending it to the device.
+    /// </summary>
+    public interface IMemoryResourceChecker : IDependency
+    {
+        /// <summary>
+        /// Verifies resource availability. If there is an issue, an exception is thrown.
+        /// </summary>
+        /// <param name="memory">The memory we want to send to the device.</param>
+        /// <param name="hardwareRepresentation">The representation of the device and program on it.</param>
+        /// <exception cref="InvalidOperationException">When there are not enough resources.</exception>

```

Comment:
Exception helyett jobb lenne inkább valami status objectet visszaadni ezekből a providerekből, és aztán exceptiont dobni csak a `MemberInvocationHandlerFactory`-ben.

Reply:
Az miért jobb?
Mi legyen ebben a status objectben?

Reply:
Azértjobb amúgy, mert ez egye ellenőrző service, az, hogy nála az ellenőrzésnél false jött ki, az nem exception. Egy hívással feljebb már lehet az, de mondjuk akkor is dönthetün úgy, hogy nem az első failnél dobunk exception, hanem akkor, ha >=1 van, és ha >1 van, akkor összegezve dobjuk.

Reply:
Cool, makes sense.